pod.go

```diff
@@ -144,4 +158,16 @@ func (pc *podCollector) collectPod(ch chan<- prometheus.Metric, p v1.Pod) {
 		addGauge(descPodContainerStatusReady, boolFloat64(cs.Ready), cs.Name)
 		addCounter(descPodContainerStatusRestarts, float64(cs.RestartCount), cs.Name)
 	}
+
+	nodeName := p.Spec.NodeName
+	for _, c := range p.Spec.Containers {
+		req := c.Resources.Requests
```

Comment:
Looking at the api definition, the comment mentions that the resources cannot change. 
The Container type  in `client-go/1.4/pkg/api/v1/types.go` has this.

```
 // Compute Resources required by this container.
 // Cannot be updated.
 // More info: http://releases.k8s.io/HEAD/docs/user-guide/persistent-volumes.md#resources
 Resources ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,8,opt,name=resources"` 
```

Are you sure the resources really belongs in the descContainerInfo ? 
One argument is about the composition of the descContainerInfo. The current tags of descContainerInfo are all from PodStatus.ContainerStatus. This new addition of ResourceRequirements is part of the PodSpec not  PodStatus. 
If we add the per container resources to descPodContainerInfo, the metric will have tags both from the PodStatus and PodSpec. I do not argue that this is a fundamental problem but may indicate a problem in the encapsulation and scope of the metrics emitted from kube-state-metrics. I wanted to double check that this is really the direction we want to go. 

A second more important concern of mine is that the labelValues in [MustNewConstMetric](https://godoc.org/github.com/prometheus/client_golang/prometheus#MustNewConstMetric)  are strings. The resources that we collect are of millicpus and bytes types. Stringifying them may be problematic for the consuming system. 

As mentioned [before](https://github.com/kubernetes/kube-state-metrics/issues/33#issue-181472679) our use case is to accumulate these request resources for all containers, in all pods running in one node. The consuming system needs to handle an atoi like conversion and then accumulation.  @fabxc gave an example code snippet [here](https://github.com/kubernetes/kube-state-metrics/issues/33#issuecomment-252182854). I am not sure whether a similar accumulation is possible if the requestedResources are stringified lables to some metric. I have taken a brief look at [prometheus operators](https://prometheus.io/docs/querying/operators/) and it was not immediately obvious to me.